internal/pkg/blob/gcs_test.go

```diff
@@ -29,103 +30,153 @@ import (
+	}
 
 	// Check to see access to this file fails now that it has been deleted.
-	got, err = gcs.Get(ctx, filePath)
-	assert.Equal(t, gcerrors.NotFound, gcerrors.Code(err),
-		"Get should return gcerrors.NotFound after object has been deleted.")
+	_, err = gcs.Get(ctx, filePath)
+	if gcerrors.Code(err) != gcerrors.NotFound {
+		t.Errorf("Get should return gcerrors.NotFound after object has been deleted, got = %v", err)
+	}
 
-	err = gcs.Delete(ctx, filePath)
-	assert.Equal(t, gcerrors.NotFound, gcerrors.Code(err),
-		"Delete should return gcerrors.NotFound when object is not found.")
+	if err = gcs.Delete(ctx, filePath); gcerrors.Code(err) != gcerrors.NotFound {
+		t.Errorf("Delete should return gcerrors.NotFound when object is not found, got = %v", err)
+	}
+}
+
+func testReader(t *testing.T, name string, rd io.ReadCloser, b []byte) {
+	t.Run(name, func(t *testing.T) {
+		if rd == nil {
+			t.Fatalf("Reader should not be nil, got = %v", rd)
+		}
+		if err := iotest.TestReader(rd, b); err != nil {
+			t.Errorf("iotest.TestReader() failed: %v", err)
+		}
+		if err := rd.Close(); err != nil {
+			t.Errorf("reader.Close() failed: %v", err)
+		}
+	})
 }
 
 func TestGCS_SimpleStreamTests(t *testing.T) {
 	t.Parallel()
 
 	ctx := context.Background()
-	gcs := getBucket(ctx, t)
+	gcs := mustGetBucket(ctx, t)
 	const filePath = "simple-stream-tests.txt"
 	testBlob := []byte("Lorem ipsum dolor sit amet, consectetur adipiscing elit")
 
 	writer, err := gcs.NewWriter(ctx, filePath)
-	require.NotNilf(t, writer, "NewWriter(%q) should not return nil.", filePath)
-	require.NoErrorf(t, err, "NewWriter(%q) should not return error.", filePath)
-
-	n, err := writer.Write(testBlob[:10])
-	assert.NoError(t, err)
-	assert.Equal(t, 10, n)
-	n, err = writer.Write(testBlob[10:])
-	assert.NoError(t, err)
-	assert.Equal(t, len(testBlob)-10, n)
-	err = writer.Close()
-	require.NoError(t, err, "writer.Close should not return error.")
-	t.Cleanup(func() { assert.NoError(t, gcs.Delete(ctx, filePath)) })
-
-	reader, err := gcs.NewReader(ctx, filePath)
-	require.NotNilf(t, writer, "NewReader(%q) should not return nil.", filePath)
-	require.NoErrorf(t, err, "NewReader(%q) should not return error.", filePath)
-
-	err = iotest.TestReader(reader, testBlob)
-	assert.NoError(t, err, "TestReader should not return error.")
-	assert.NoError(t, reader.Close())
+	if writer == nil {
+		t.Fatalf("NewWriter(%q) should not return nil, got = %v", filePath, writer)
+	}
+	if err != nil {
+		t.Fatalf("NewWriter(%q) should not fail, got = %v", filePath, err)
+	}
+
+	for _, tc := range []struct {
+		start, end, want int
+	}{
+		{0, 10, 10},
+		{10, len(testBlob), len(testBlob) - 10},
+	} {
+		t.Run(fmt.Sprintf("Write [%v:%v]", tc.start, tc.end), func(t *testing.T) {
+			got, err := writer.Write(testBlob[tc.start:tc.end])
+			if err != nil {
+				t.Errorf("Write() failed: %v", err)
+			}
+			if tc.want != got {
+				t.Errorf("Write() = want %v, got %v", tc.want, got)
```

Comment:
Prefer order of actual (got) and then expected (want) rather than other way around 

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures